jerry-debugger/jerry-client-ws.py

```diff
@@ -0,0 +1,676 @@
+
+        while len(result) < len_expected:
+            result += self.client_socket.recv(1024)
+
+        len_result = len(result)
+
+        if (ord(result[0]) != 0x82
+            or ord(result[1]) != 4
+            or ord(result[2]) != JERRY_DEBUGGER_CONFIGURATION):
+            raise Exception('Unexpected configuration')
+
+        self.max_message_size = ord(result[3])
+        self.cp_size = ord(result[4])
+        self.little_endian = ord(result[5])
+
+        if self.little_endian:
+            self.byte_order = '<'
+            logging.debug('Little endian machine')
+        else:
+            self.byte_order = '>'
+            logging.debug('Big endian machine')
+
+        if self.cp_size == 2:
+            self.cp_format = 'H'
+        else:
+            self.cp_format = 'I'
+
+        self.idx_format = 'I'
+
+        logging.debug('Compressed pointer size: %d' % (self.cp_size))
+
+        if len_result > len_expected:
+            self.message_data = result[len_expected:]
+
+    def __del__(self):
+        self.client_socket.close()
+
+    def send_message(self, message):
+        size = len(message)
+        while size > 0:
+            bytes_send = self.client_socket.send(message)
+            if bytes_send < size:
+                message = message[bytes_send:]
+            size -= bytes_send
+
+    def send_breakpoint(self, breakpoint):
+        message = pack(self.byte_order + 'BBIBB' + self.cp_format + self.idx_format,
+                       0x82,
+                       0x80 + 1 + 1 + self.cp_size + 4,
+                       0,
+                       JERRY_DEBUGGER_UPDATE_BREAKPOINT,
+                       int(breakpoint.active_index >= 0),
+                       breakpoint.function.byte_code_cp,
+                       breakpoint.offset)
+        self.send_message(message)
+
+    def send_command(self, command):
+        message = pack(self.byte_order + 'BBIB',
+                       0x82,
+                       0x80 + 1,
+                       0,
+                       command)
+        self.send_message(message)
+
+    def get_message(self):
+        if self.message_data is None:
+            return None
+
+        while True:
+            if len(self.message_data) >= 2:
+                if ord(self.message_data[0]) != 0x82:
+                    raise Exception('Unexpected data frame')
+
+                size = ord(self.message_data[1])
+                if size == 0 or size >= 126:
+                    raise Exception('Unexpected data frame')
+
+                if len(self.message_data) >= size + 2:
+                    result = self.message_data[0:size + 2]
+                    self.message_data = self.message_data[size + 2:]
+                    return result
+
+            data = self.client_socket.recv(MAX_BUFFER_SIZE)
+            if not data:
+                self.message_data = None
+                return None
+
+            self.message_data += data
+
+
+def parse_source(debugger, data):
+
```

Comment:
Sorry, misunderstand, I thought you point for the 412 & 413 blank line, not for the 415.